tests/subsys/settings/fcb/src/settings_test_fcb.c

```diff
@@ -341,66 +341,50 @@ int c3_handle_export(int (*cb)(const char *name,
 	return 0;
 }
 
-void test_settings_check_target(void)
+void *settings_config_fcb_setup(void)
 {
 	const struct flash_area *fap;
 	int rc;
 	uint8_t wbs;
 
 	rc = flash_area_open(FLASH_AREA_ID(storage), &fap);
-	zassert_true(rc == 0, "Can't open storage flash area");
+	zassume_true(rc == 0, "Can't open storage flash area");
 
 	wbs = flash_area_align(fap);
-	zassert_true(wbs <= 16,
+	zassume_true(wbs <= 16,
 		"Flash driver is not compatible with the settings fcb-backend");
+	return NULL;
 }
 
-void test_settings_encode(void);
-void test_config_empty_lookups(void);
-void test_config_insert(void);
-void test_config_getset_unknown(void);
-void test_config_getset_int(void);
-void test_config_getset_int64(void);
-void test_config_commit(void);
-
-void test_config_empty_fcb(void);
 void test_config_save_1_fcb(void);
+void test_config_save_fcb_unaligned(void);
 void test_config_delete_fcb(void);
 void test_config_insert2(void);
 void test_config_save_2_fcb(void);
+void test_config_compress_reset(void);
 void test_config_insert3(void);
 void test_config_save_3_fcb(void);
-void test_config_compress_reset(void);
-void test_config_save_one_fcb(void);
-void test_config_compress_deleted(void);
-void test_setting_raw_read(void);
-void test_setting_val_read(void);
-void test_config_save_fcb_unaligned(void);
 
-void test_main(void)
+ZTEST(settings_config_fcb, test_config_handler1)
+{
+	test_config_save_1_fcb();
+	test_config_save_fcb_unaligned();
+	test_config_delete_fcb();
+}
+
+ZTEST(settings_config_fcb, test_config_handler2)
+{
+	test_config_insert2();
+	test_config_save_2_fcb();
+	test_config_compress_reset();
+}
```

Comment:
These seem to be order dependent based on the name of the test in ZTEST. This is not supported behavior. I verified this by downloading and testing the PR.

It looks like the legacy test was highly dependent on the test order, meaning it was really just one large test. That's a problem with the test itself. If that's the behavior that is desired here then just wrap all test calling functions in a single ZTEST.

The order of which ZTEST test gets executed should never be depended upon.

Reply:
Thank you for your verification @aaronemassey !
Updated, test cases in this file can be run with CONFIG_ZTEST_SHUFFLE now.

Reply:
So these are no longer order-dependent?

Reply:
sure, tests are no longer order-dependent.